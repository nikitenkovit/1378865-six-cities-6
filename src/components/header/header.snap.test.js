import React from "react";
import {render} from "@testing-library/react";
import Header from "./header";
import {Provider} from "react-redux";
import {Router} from "react-router-dom";
import browserHistory from "../../history";
import configureStore from "redux-mock-store";

const mockStore = configureStore({});

const wrapper = ({children}) => (
  <Provider store={mockStore({
    USER: {
      email: `test@test.ru`,
      name: `Name`,
    }
  })}>
    <Router history={browserHistory}>
      {children}
    </Router>
  </Provider>
);

it(`Should Header render correctly`, () => {
  const {container} = render(<Header isFavoriteScreen={true} />, {wrapper});
  expect(container).toMatchSnapshot();
});
