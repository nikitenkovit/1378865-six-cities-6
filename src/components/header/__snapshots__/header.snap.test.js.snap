// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Header render correctly 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="container"
    >
      <div
        class="header__wrapper"
      >
        <div
          class="header__left"
        >
          <a
            class="header__logo-link header__logo-link--active"
            href="/"
          >
            <img
              alt="6 cities logo"
              class="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          class="header__nav"
        >
          <ul
            class="header__nav-list"
            style="flex-direction: column; align-items: flex-end;"
          >
            <li
              class="header__nav-item user"
            >
              <a
                class="header__nav-link header__nav-link--profile"
                href="/favorites"
              >
                <span
                  class="header__user-name user__name"
                >
                  Sign in
                </span>
              </a>
            </li>
            <li
              class="header__nav-item user"
              style="margin-left: 15px;"
            >
              <a
                class="header__nav-link header__nav-link--profile"
              >
                <span
                  class="header__user-name user__name"
                >
                  Logout
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
</div>
`;
